service/tablet_allocator.cc

```diff
@@ -264,6 +264,75 @@ class load_balancer {
         }
     };
 
+    // Maximum tablet size works as the split threshold, whereas minimum size works as the
+    // merge threshold. To prevent oscillations between split and merge, hysteresis is applied
+    // to the decision.
+    // TODO: possibly make target_max_tablet_size a config to help writing tests that stresses resize path.
+    static const size_t target_max_tablet_size = 10UL * 1024 * 1024 * 1024;
+    static const size_t target_min_tablet_size = double(target_max_tablet_size / 2) * 0.6;
+
+    // This limits defines the maximum number of resize requests happening at a point in time.
+    // Due to power-of-two constraint, resize is always a cluster-wide operation meaning that
+    // a table will have all of its tablets resized, not only a subset. Therefore, it's important
+    // to set a limit on the operation, to avoid interfering with the foreground activity.
+    static const size_t max_concurrent_resize_requests = 1;
```

Comment:
Don't agree, if we have many small tables they'll need to be resized concurrently.

However, let's not add complications.

Reply:
btw this won't increase interference, since all of this is happening in the maintenance group

Reply:
You're right about interference with foreground, but it's still bad to cause interference within maintenance group, right? We don't want many concurrent splits to slow down more (or equally) important maintenance tasks. We don't have nested groups yet.

Reply:
Don't see why splitting isn't important, it controls the speed at which we can scale.

Also consider that some splits will leave some nodes/shards unutilized, so concurrent splits can get more of the cluster to work.

Reply:
Makes sense. But replicas still throttle split compactions (at compaction manager level) as they don't have unlimited bandwidth. Right now, split compaction is serialized like any other maintenance task, but we can definitely change that. It seems you don't care really about stable clusters, as they aren't supposed to have frequency split requests anyway. Another thing that will be helpful is using growth factor of 4 instead of 2.

Now that I put more thought on it, the need for split of each table is still determined by how fast replicas can write, so it's not like coordinator will throw at replicas more work than they can handle. 

I am now inclined to removing this concurrency restriction (at coordinator) and let each replica decide the amount of splitting work they can handle at any point in time. Do you agree?